src/library/itunes/itunesxmlimporter.h

```diff
@@ -0,0 +1,40 @@
+#pragma once
+
+#include <QSqlDatabase>
+#include <QXmlStreamReader>
+#include <atomic>
+#include <memory>
+
+#include "library/itunes/itunesimporter.h"
+#include "library/itunes/itunespathmapping.h"
+#include "library/libraryfeature.h"
+
+/// An importer that parses an iTunes XML library.
+class ITunesXMLImporter : public ITunesImporter {
+  public:
+    ITunesXMLImporter(LibraryFeature* parentFeature,
+            const QString& filePath,
+            QSqlDatabase& database,
+            ITunesPathMapping& pathMapping,
+            std::atomic<bool>& cancelImport);
+
+    ITunesImport importLibrary() override;
+
+  private:
+    LibraryFeature* m_parentFeature;
+    QFile m_file;
+    QXmlStreamReader m_xml;
+    QString m_dbfile;
+    QSqlDatabase& m_database;
+    ITunesPathMapping& m_pathMapping;
+    std::atomic<bool>& m_cancelImport;
```

Comment:
My internal, Rust-conditioned borrow checker is tilting if I see such a construct.

Wrapping the atomic into a shared_ptr would be much safer. The allocation only happens once during a session,

Reply:
> My internal, Rust-conditioned borrow checker is tilting if I see such a construct.

Just imagine the importer being parameterized over `<'a>` and turn all of these into `&'a` 😄 

> Wrapping the atomic into a shared_ptr would be much safer

That what one solution I suggested in https://github.com/mixxxdj/mixxx/pull/11353#discussion_r1138980135 which was rejected though.

Reply:
@Swiftb0y I don't understand your comment on the reference count? This must always be atomic like in an `Arc` (C++ does not have `Rc`) and doesn't require additional wrapping.

Sure, I am aware that the importers are scoped somehow. But this scope or context is not modeled explicitly anywhere. I would expect at least a comment that explains these implicit assumptions.

Reply:
> I don't understand your comment on the reference count? This must always be atomic like in an Arc (C++ does not have Rc) and doesn't require additional wrapping.

Ah yes, you are correct, the shared_ptr docs on this are a bit hard to understand. Disregard my comment on `std::atomic<std::shared_ptr>` then. 

> Sure, I am aware that the importers are scoped somehow. But this scope or context is not modeled explicitly anywhere. I would expect at least a comment that explains these implicit assumptions.

Yes, a comment makes sense. I agree its not great, but its what we currently have. We already agreed-to solution of implementing virtual cancel member function solves the lifetime problem (in addition to problems I outlined when we had discussed them).

Reply:
Don't get me wrong, using a `std::shared_ptr` is fine for me. But throwing out the assumption that the `*Importer` doesn't outlive the `ITunesFeature` will cause even more problems.

Reply:
So should we go with `shared_ptr` or keep it the current way?

Reply:
Add a comment to the member. For anyone who is not aware of the context when stumbling over this code.

Reply:
@Swiftb0y So using mutable references is now allowed? I remember that there was a _pointer-only policy_ for mutable references.

Reply:
True, I often forget that. we can make `m_cancelImport` a const reference. Or pointer, I don't care either way. Nonetheless, I struggle to understand why we are still discussing details of this sub-par design decision we already agreed to replace in a future PR.

Reply:
Using a const-ref would be sufficient here.

Reply:
The value is read-only for the subordinate components.

Reply:
The C++ atomic semantics differ from Rust. Without interior mutability atomics would not be useful in Rust.

Reply:
I've updated these to `const` references where possible and added a small note on lifetimes in fb3cc2ab325e99fd78acffe3691d0403be8c0852